bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageFileProvider.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2025 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Hannes Wellmann - introduce ImageFileProvider as replacement for ImageFileNameProvider
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.nio.file.Path;
+
+/**
+ * Interface to provide a callback mechanism to get information about images
+ * when the application is moved from a low DPI monitor to a high DPI monitor.
+ * This provides API which will be called by SWT during the image rendering.
+ *
+ * This interface needs to be implemented by client code to provide the image
+ * information on demand.
+ *
+ * @since 3.130
+ */
+@FunctionalInterface
+public interface ImageFileProvider {
```

Comment:
Shouldn't this be `ImagePathProvider`? Mixing up terms "file" and "path" may be confusing.